### PR TITLE
Added a --format=json option to the analyze command (CRAFT-320)

### PR DESCRIPTION
**Pull request by facundobatista**

### CODE REVIEW

File: charmcraft/commands/analyze.py [hunk 1]
```diff
@@ -75,6 +83,20 @@ def run(self, parsed_args):
...
+            info = [
+                {
+                    "name": r.name,
+                    "result": r.result,
+                    "url": r.url,
+                    "type": r.check_type,
+                }
+                for r in linting_results
+            ]
+            logger.info(json.dumps(info))
```
Comment (sergiusens) [inline charmcraft/commands/analyze.py:97]: wouldn't we preffer to pretty print the output? `indent=4` to `json.dumps`

**Review by sergiusens — APPROVED** [review]
Looks good, I would advise to pretty print the output even if it is machine oriented.

File: charmcraft/commands/analyze.py [hunk 2]
```diff
@@ -54,6 +57,11 @@ def fill_parser(self, parser):
             action="store_true",
             help="Force to run all checks, even those set to ignore in the configuration",
         )
+        parser.add_argument(
+            "--format",
+            choices=[JSON_FORMAT],
```
Comment (cmatsuoka) [inline charmcraft/commands/analyze.py:62]: Will we offer other formats in the future?

**Review by cmatsuoka — APPROVED** [review]
Looks good.

File: charmcraft/commands/analyze.py (hunk 2 above)
Comment (facundobatista) [reply charmcraft/commands/analyze.py:62]: Maybe! No plans by now, though.

File: charmcraft/commands/analyze.py (hunk 1 above)
Comment (facundobatista) [reply charmcraft/commands/analyze.py:97]: Ooooooh, very nice!! Used it!

**Pull request merged**